api/v1alpha1/nutanix_node_types.go

```diff
@@ -175,3 +187,38 @@ func (NutanixResourceIdentifiers) VariableSchema() clusterv1.VariableSchema {
 		},
 	}
 }
+
+type NutanixCategoryIdentifier capxv1.NutanixCategoryIdentifier
+
+func (NutanixCategoryIdentifier) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Nutanix Category Identifier",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"key": {
+					Type:        "string",
+					Description: "key is the Key of category in PC.",
+				},
+				"value": {
+					Type:        "string",
+					Description: "value is the category value linked to the category key in PC.",
+				},
```

Comment:
Let's add validation constraints via format and/or pattern

Reply:
I can find no limitations on either key or value.

- https://www.nutanixbible.com/classic#categories